File: go/greeting_activities.go
package hello

import "context"

// String-composition activities (the greeting demos).

// ComposeGreeting is the Hello World activity. Breakpoint here for the clean Lab 3
// pause — activities have no deadlock detector and no workflow-task timeout.
func ComposeGreeting(ctx context.Context, in GreetingInput) (string, error) {
	// >>> BREAKPOINT (activity code): recommended pause point. <<<
	return "Hello, " + in.Name + "!", nil
}

// ComposeFarewell returns a farewell string.
func ComposeFarewell(ctx context.Context, in GreetingInput) (string, error) {
	// >>> BREAKPOINT (activity) <<<
	return "Goodbye, " + in.Name + "!", nil
}

File: python/greeting_activities.py
"""String-composition activities (the greeting demos)."""

from temporalio import activity

from shared import GreetingInput


@activity.defn
async def compose_greeting(inp: GreetingInput) -> str:
    # >>> BREAKPOINT (activity) <<<
    return f"Hello, {inp.name}!"


@activity.defn
async def compose_farewell(inp: GreetingInput) -> str:
    # >>> BREAKPOINT (activity) <<<
    return f"Goodbye, {inp.name}!"
